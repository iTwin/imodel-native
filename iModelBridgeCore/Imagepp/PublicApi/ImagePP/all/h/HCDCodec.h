//:>--------------------------------------------------------------------------------------+
//:>
//:>     $Source: PublicApi/ImagePP/all/h/HCDCodec.h $
//:>
//:>  $Copyright: (c) 2015 Bentley Systems, Incorporated. All rights reserved. $
//:>
//:>+--------------------------------------------------------------------------------------
//-----------------------------------------------------------------------------
// Class : HCDCodec
//-----------------------------------------------------------------------------
// General class for codecs.
//-----------------------------------------------------------------------------

#pragma once

#include <Imagepp/all/h/HFCExclusiveKey.h>

BEGIN_IMAGEPP_NAMESPACE

class HCDPacketRLE;

// Binary optimization: Speed up read/write when codec support direct decompression/compression to/from RLE format.
class HNOVTABLEINIT HCDCodecRLEInterface
    {
public:
    virtual void     DecompressSubsetToRLE(const void* pi_pInData, size_t pi_InDataSize, HFCPtr<HCDPacketRLE>& pio_rpRLEPacket) = 0;
    virtual size_t   CompressSubsetFromRLE(HFCPtr<HCDPacketRLE> const& pi_rpPacketRLE, void* po_pOutBuffer, size_t po_OutBufferSize) = 0;
    };

class HNOVTABLEINIT HCDCodec : public HFCShareableObject<HCDCodec>,
    public HFCExclusiveKey     // to have access to claim/release instead of including m_Key and adding the methods
    {
    HDECLARE_BASECLASS_ID(HCDCodecId_Base)

public:

    // codec states enumeration
    enum State
        {
        STATE_NONE,
        STATE_COMPRESS,
        STATE_DECOMPRESS
        };

    // destructor
    virtual         ~HCDCodec();

    // compression
    virtual size_t          CompressSubset(const void* pi_pInData,
                                           size_t pi_InDataSize,
                                           void* po_pOutBuffer,
                                           size_t pi_OutBufferSize) = 0;

    virtual size_t          DecompressSubset(const void* pi_pInData,
                                             size_t pi_InDataSize,
                                             void* po_pOutBuffer,
                                             size_t pi_OutBufferSize) = 0;

    State                   GetCurrentState() const;
    virtual void            Reset();

    virtual HCDCodec*       Clone() const = 0;

    virtual size_t          GetDataSize() const = 0;

    // subset
    virtual size_t          GetMinimumSubsetSize() const = 0;
    virtual size_t          GetSubsetMaxCompressedSize() const = 0;
    virtual void            SetSubsetSize(size_t pi_Size) = 0;

    // others
    virtual double          GetEstimatedCompressionRatio() const;
    virtual bool            HasRandomAccess() const;
    virtual bool            IsCodecImage() const;
    virtual bool            IsCodecVector() const;
    virtual const WString&  GetName() const;

    virtual HCDCodecRLEInterface* 
                            GetRLEInterface();

protected:

    // constructors
    HCDCodec(const WString& pi_rCodecName);
    HCDCodec(const HCDCodec& pi_rObj);


    // compression
    void                    SetCurrentState(State pi_State);

    WString         m_CodecName;

private:

    State          m_CurrentState;
    };

END_IMAGEPP_NAMESPACE