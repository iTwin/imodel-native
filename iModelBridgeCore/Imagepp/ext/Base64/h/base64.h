#ifndef _base64_h_
#define _base64_h_

#include <iostream>

namespace b64
{
        bool encode(std::istream& input, std::ostream& output);
        bool decode(std::istream& input, std::ostream& output);
};

#endif
