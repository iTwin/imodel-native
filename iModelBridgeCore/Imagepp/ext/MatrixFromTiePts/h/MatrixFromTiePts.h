/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See COPYRIGHT.md in the repository root for full copyright notice.
*--------------------------------------------------------------------------------------------*/

#pragma once


#define TRANSFO_NONE     -1 /* No transformation model */
#define TRANSFO_TRAN_2D   0 /* Translation en x et y */
#define TRANSFO_SIMI_2D   1 /* Translation x,y + Rotation + Echelle */
#define TRANSFO_POLY_2D_1 2 /* Polynomiale deg 1 */
#define TRANSFO_AFFI_2D   3 /* Tran. x,y + Rotation + Echelle + Axes */
#define TRANSFO_PROJ_2D   4 /* Projection d'un plan sur un autre */
#define TRANSFO_POLY_2D_2 5 /* Polynomiale deg 2 */
#define TRANSFO_POLY_2D_3 6 /* Polynomiale deg 3 */
#define TRANSFO_POLY_2D_4 7 /* Polynomiale deg 4 */ /* A tester ??? */
#define TRANSFO_POLY_2D_5 8 /* Polynomiale deg 5 */ /* A tester ??? */
#define TRANSFO_MAT_2D    9 /* Matrice de transformation */
#define TRANSFO_HELM_2D   10
#define TRANSFO_TPS_2D    11
#define TRANSFO_MAX       12

#define TRF_ERROR_MAX_ITERATIONS    -1
#define TRF_ERROR_NO_MEMORY         -2
#define TRF_ERROR_BAD_NUMBER_POINTS -3
#define TRF_ERROR_MATRIX            -4


long GetTransfoMatrixFromScaleAndTiePts (double          po_pMatrix[4][4],
                                         unsigned short  pi_NbVal_GeoTiePoint,
                                         double*         pi_pVal_GeoTiePoint,
                                         unsigned short  pi_NbVal_GeoPixelScale,
                                         double*         pi_pVal_GeoPixelScale);

long GetProjectiveTransfoMatrixFromScaleAndTiePts(double         po_pMatrix[4][4],
                                                  unsigned short pi_NbVal_GeoTiePoint,
                                                  double*        pi_pVal_GeoTiePoint);

long GetAffineTransfoMatrixFromScaleAndTiePts(double         po_pMatrix[4][4],
                                              unsigned short pi_NbVal_GeoTiePoint,
                                              double*        pi_pVal_GeoTiePoint);

long GetTransfoMatrixFromTiePts(double           po_pMatrix[4][4],
                                unsigned short   pi_NbVal_GeoTiePoint,
                                const double*    pi_pVal_GeoTiePoint,
                                long             pi_ModelType);

