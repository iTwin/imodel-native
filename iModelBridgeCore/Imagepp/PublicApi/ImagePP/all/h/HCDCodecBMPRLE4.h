//:>--------------------------------------------------------------------------------------+
//:>
//:>
//:>  Copyright (c) Bentley Systems, Incorporated. All rights reserved.
//:>
//:>+--------------------------------------------------------------------------------------
// Class : HCDCodecBMPRLE8
//-----------------------------------------------------------------------------
// RLE8 codec lass.
//-----------------------------------------------------------------------------
#pragma once

#include "HCDCodecImage.h"

BEGIN_IMAGEPP_NAMESPACE

class HCDCodecBMPRLE4 : public HCDCodecImage
    {
    HDECLARE_CLASS_ID(HCDCodecId_BMPRLE4, HCDCodecImage)

public:

    IMAGEPP_EXPORT                 HCDCodecBMPRLE4();

    HCDCodecBMPRLE4(size_t pi_Width,
                    size_t pi_Height);

    HCDCodecBMPRLE4(const HCDCodecBMPRLE4& pi_rObj);

    ~HCDCodecBMPRLE4();

    size_t          CompressSubset(const void* pi_pInData,
                                   size_t     pi_InDataSize,
                                   void*      po_pOutBuffer,
                                   size_t     pi_OutBufferSize) override;

    size_t          DecompressSubset(const void* pi_pInData,
                                     size_t       pi_InDataSize,
                                     void*        po_pOutBuffer,
                                     size_t       pi_OutBufferSize) override;

    bool           HasLineAccess() const override;

    virtual HCDCodec* Clone() const override;

    bool           IsBitsPerPixelSupported(size_t pi_Bits) const override;

    size_t          GetSubsetMaxCompressedSize() const override;

protected:

private:
    };

END_IMAGEPP_NAMESPACE
