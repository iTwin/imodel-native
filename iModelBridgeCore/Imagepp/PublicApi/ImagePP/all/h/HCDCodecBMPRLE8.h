//:>--------------------------------------------------------------------------------------+
//:>
//:>     $Source: PublicApi/ImagePP/all/h/HCDCodecBMPRLE8.h $
//:>
//:>  $Copyright: (c) 2015 Bentley Systems, Incorporated. All rights reserved. $
//:>
//:>+--------------------------------------------------------------------------------------
// Class : HCDCodecBMPRLE8
//-----------------------------------------------------------------------------
// RLE8 codec lass.
//-----------------------------------------------------------------------------
#pragma once

#include "HCDCodecImage.h"

BEGIN_IMAGEPP_NAMESPACE

class HCDCodecBMPRLE8 : public HCDCodecImage
    {
    HDECLARE_CLASS_ID(HCDCodecId_BMPRLE8, HCDCodecImage)

public:

    IMAGEPP_EXPORT                 HCDCodecBMPRLE8();

    HCDCodecBMPRLE8(  size_t pi_Width,
                      size_t pi_Height);

    HCDCodecBMPRLE8(const HCDCodecBMPRLE8& pi_rObj);

    ~HCDCodecBMPRLE8();

    size_t          CompressSubset(const void* pi_pInData,
                                   size_t pi_InDataSize,
                                   void* po_pOutBuffer,
                                   size_t pi_OutBufferSize);

    size_t          DecompressSubset(const void* pi_pInData,
                                     size_t pi_InDataSize,
                                     void* po_pOutBuffer,
                                     size_t pi_OutBufferSize);

    bool           HasLineAccess() const;

    virtual HCDCodec* Clone() const override;

    bool           IsBitsPerPixelSupported(size_t pi_Bits) const;

    size_t          GetSubsetMaxCompressedSize() const;

protected:

private:
    };

END_IMAGEPP_NAMESPACE