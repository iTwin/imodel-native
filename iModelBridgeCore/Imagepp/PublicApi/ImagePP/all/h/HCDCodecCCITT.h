//:>--------------------------------------------------------------------------------------+
//:>
//:>
//:>  Copyright (c) Bentley Systems, Incorporated. All rights reserved.
//:>
//:>+--------------------------------------------------------------------------------------
//-----------------------------------------------------------------------------
// Class : HCDCodecCCITT
//-----------------------------------------------------------------------------
// CCITT codec lass.
//-----------------------------------------------------------------------------

#pragma once

#include "HCDCodecImage.h"

BEGIN_IMAGEPP_NAMESPACE

class HCDCodecCCITT : public HCDCodecImage
    {
    HDECLARE_CLASS_ID(HCDCodecId_CCITT, HCDCodecImage)

public:

    ~HCDCodecCCITT();

    bool           IsBitsPerPixelSupported(size_t pi_Bits) const override;

    HCDCodecCCITT(const HCDCodecCCITT& pi_rObj);

    size_t          GetSubsetMaxCompressedSize() const override;

    double GetEstimatedCompressionRatio() const override;

    IMAGEPP_EXPORT void     SetPhotometric(uint16_t pi_Photo);
    uint16_t GetPhotometric() const;

    IMAGEPP_EXPORT void     SetBitRevTable(bool pi_Reverse);

protected:
    
    virtual HCDCodec* Clone() const override = 0;

    HCDCodecCCITT();

    HCDCodecCCITT(size_t pi_Width,
                  size_t pi_Height);

    bool           m_bitrevtable;
    uint16_t m_photometric;
    };

END_IMAGEPP_NAMESPACE
