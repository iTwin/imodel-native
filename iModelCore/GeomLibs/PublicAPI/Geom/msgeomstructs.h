/*--------------------------------------------------------------------------------------+
|
|     $Source: PublicAPI/Geom/msgeomstructs.h $
|
|  $Copyright: (c) 2011 Bentley Systems, Incorporated. All rights reserved. $
|
+--------------------------------------------------------------------------------------*/
#pragma once

/*__BENTLEY_INTERNAL_ONLY__*/

#include <Geom/GeomApi.h>


