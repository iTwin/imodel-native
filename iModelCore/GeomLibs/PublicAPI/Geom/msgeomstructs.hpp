/*--------------------------------------------------------------------------------------+
|
|     $Source: PublicAPI/Geom/msgeomstructs.hpp $
|
|  $Copyright: (c) 2009 Bentley Systems, Incorporated. All rights reserved. $
|
+--------------------------------------------------------------------------------------*/
#pragma once
/*__BENTLEY_INTERNAL_ONLY__*/
#include <Geom/GeomApi.h>
