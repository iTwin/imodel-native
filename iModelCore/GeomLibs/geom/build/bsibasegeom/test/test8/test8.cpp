/*--------------------------------------------------------------------------------------+
|
|  Copyright (c) Bentley Systems, Incorporated. All rights reserved.
|
+--------------------------------------------------------------------------------------*/
// test1.cpp : Defines the entry point for the console application.
//

#include <stdio.h>
#include <msgeomstructs.hpp>
#include <msgeomstructs.h>
#include <printfuncs.h>
#include <stdlib.h>


void test1_DRange2d ()
    {
    DRange2d rangeA, rangeB, rangeC, rangeD;
    rangeA.init ();
    rangeB.init ();
    checkDRange2d (&rangeA, &rangeB, "init");
    rangeA.initFrom (1,2,2.5,4.3);
    checkDRange2d (&rangeA, &rangeA, "initFrom (xyxy)");
    checkTrue  (rangeB.isNull (), "DRange2d.isNull()");
    checkFalse (rangeA.isNull (), "DRange2d.isNull()");

    rangeA.initFrom (1,2,3,5);
    checkDouble (rangeA.extentSquared (), 13.0, "DRange2d.extentSquared");
    checkDouble (rangeA.area (), 6.0, "DRange2d.area");
    checkFalse (rangeA.isPoint (), "DRange2d.isPoint ()");

    checkFalse (rangeA.isEmpty (), "DRange2d.isEmpty()");
    rangeA.initFrom (1,2);
    checkTrue (rangeA.isPoint (), "DRange2d.isPoint ()");
    checkFalse (rangeA.isEmpty (), "DRange2d.isEmpty()");

    rangeA.high.x -= 0.2;
    checkTrue (rangeA.isEmpty (), "DRange2d.isEmpty() on x");
    rangeA.initFrom (1,2);
    rangeA.high.x -= 0.2;
    checkTrue (rangeA.isEmpty (), "DRange2d.isEmpty() on y");

    DPoint2d pointQ, pointR;
    pointQ.x = 3.0;
    pointQ.y = 4.0;
    pointR.x = 1.0;
    pointR.y = 2.3;
    rangeA.initFrom (&pointQ);
    rangeB.initFrom(pointQ.x, pointQ.y);
    checkDRange2d (&rangeA, &rangeB, "DRange2d::initFrom(x,y)==initFrom(point)");

    rangeA.initFrom(&pointQ, &pointR);
    rangeB.initFrom (&pointR, &pointQ);
    checkDRange2d (&rangeA, &rangeB, "DRange2d::initFrom (P,Q) order");

    rangeC.initFrom (pointQ.x, pointQ.y, pointR.x, pointR.y);
    rangeD.initFrom (pointR.x, pointR.y, pointQ.x, pointQ.y);
    checkDRange2d (&rangeC, &rangeD, "DRange2d::initFrom(x,y,x,y) order");
    checkDRange2d (&rangeA, &rangeC, "DRange2d::initFrom");

    rangeC.init ();
    rangeD.init ();

    pointQ.x += 3.1;
    pointQ.y += 0.2;
    rangeC.initFrom (&pointQ);
    rangeD.initFrom( &pointQ, &pointQ);
    checkDRange2d (&rangeC, &rangeD, "DRange2d init from same point");

    for (double theta0 = -3; theta0 < 4.0; theta0 += 1.0)
        {
        for (double sweep = -6.25; sweep < 6.28; sweep += 1.5)
            {
            rangeA.initFromUnitArcSweep (theta0, sweep);
            rangeB.init ();
            int numStep = 32;
            double df = 1.0 / (double)numStep;
            for (int i = 0; i <= numStep; i++)
                {
                double f = i * df;
                double theta = theta0 + f * sweep;
                double x = cos (theta);
                double y = sin (theta);
                rangeB.extend (x, y);
                }
            double areaA = rangeA.area ();
            double areaB = rangeB.area ();
            checkTrue (rangeB.isContained (&rangeA), "Arc Containment");
            checkTrue (areaB <= areaA && areaB > 0.90 * areaA,
                    "unit arc sweep range");
            }
        }
    }

void test2_DRange2d ()
    {
    double xx[5] = {1,2,3,4,5};
    double yy[5] = {1.5, 2.5, 3.5, 4.5, 5.5};

    DRange2d range;

    range.initFrom (xx[1], yy[1], xx[3], yy[3]);

    for (int i = 0; i < 5; i++)
        {
        for (int j = 0; j < 5; j++)
            {
            char message1[128];
            char message2[128];
            char message3[128];
            sprintf (message1, "Contains(x,y)%d%d", i, j);
            sprintf (message2, "Contains(xy)%d%d", i, j);
            sprintf (message3, "Contains(xyz)%d%d", i, j);
            DPoint2d xy;
            xy.x = xx[i];
            xy.y = yy[j];
            DPoint3d xyz;
            xyz.x = xx[i];
            xyz.y = yy[j];
            xyz.z = 0.0;
            if (   i >= 1
                && i <= 3
                && j >= 1
                && j <= 3
                )
                {
                checkTrue  (range.contains (xx[i], yy[j]), message1);
                checkTrue  (range.contains (&xy),  message2);
                checkTrue  (range.contains (&xyz), message3);
                }
            else
                {
                checkFalse (range.contains (xx[i], yy[j]), message1);
                checkFalse  (range.contains (&xy),  message2);
                checkFalse  (range.contains (&xyz), message3);
                }
            }
        }
    }

int main(int argc, char * argv[])
    {
    initErrorTracking (NULL, argc, argv);
    test1_DRange2d ();
    test2_DRange2d ();
    return getExitStatus();
    }
