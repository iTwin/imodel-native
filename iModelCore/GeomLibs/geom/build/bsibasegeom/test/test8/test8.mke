#--------------------------------------------------------------------------------------
#
#  Copyright (c) Bentley Systems, Incorporated. All rights reserved.
#
#--------------------------------------------------------------------------------------
TESTAPP_NAME=test8

%include    $(_MakeFilePath)/../testapp.mki
