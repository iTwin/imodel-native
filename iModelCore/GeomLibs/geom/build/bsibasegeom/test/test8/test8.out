 geomlibs test environment -- command line arguments:
 Tail of argv[0] = bsigeomtest8.exe
<OK name="init"><DRange2d::NULL/></OK>
<OK name="initFrom (xyxy)"><DRange2d>
    <DPoint2d name="low">1,2</DPoint2d>
    <DPoint2d name="high">2.5,4.3</DPoint2d></DRange2d></OK>
<OK name="DRange2d.isNull()"><bool::true/></OK>
<OK name="DRange2d.isNull()"><bool::false/></OK>
<OK name="DRange2d.extentSquared"><double>13</double></OK>
<OK name="DRange2d.area"><double>6</double></OK>
<OK name="DRange2d.isPoint ()"><bool::false/></OK>
<OK name="DRange2d.isEmpty()"><bool::false/></OK>
<OK name="DRange2d.isPoint ()"><bool::true/></OK>
<OK name="DRange2d.isEmpty()"><bool::false/></OK>
<OK name="DRange2d.isEmpty() on x"><bool::true/></OK>
<OK name="DRange2d.isEmpty() on y"><bool::true/></OK>
<OK name="DRange2d::initFrom(x,y)==initFrom(point)"><DRange2d>
    <DPoint2d name="low">3,4</DPoint2d>
    <DPoint2d name="high">3,4</DPoint2d></DRange2d></OK>
<OK name="DRange2d::initFrom (P,Q) order"><DRange2d>
    <DPoint2d name="low">1,2.3</DPoint2d>
    <DPoint2d name="high">3,4</DPoint2d></DRange2d></OK>
<OK name="DRange2d::initFrom(x,y,x,y) order"><DRange2d>
    <DPoint2d name="low">1,2.3</DPoint2d>
    <DPoint2d name="high">3,4</DPoint2d></DRange2d></OK>
<OK name="DRange2d::initFrom"><DRange2d>
    <DPoint2d name="low">1,2.3</DPoint2d>
    <DPoint2d name="high">3,4</DPoint2d></DRange2d></OK>
<OK name="DRange2d init from same point"><DRange2d>
    <DPoint2d name="low">6.1,4.2</DPoint2d>
    <DPoint2d name="high">6.1,4.2</DPoint2d></DRange2d></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
<OK name="Arc Containment"><bool::true/></OK>
<OK name="unit arc sweep range"><bool::true/></OK>
Exit with 0 errors, 143 OK
